### PR TITLE
model: add TimeSformer video encoders

### PR DESCRIPTION
**Pull request by hubielu**
Three TimeSformer checkpoints: timesformer-base-finetuned-k400, timesformer-base-finetuned-ssv2, timesformer-hr-finetuned-k400. Split out of #5094 at @Samoed's suggestion.

Drafted with Claude; I ran and verified everything below.

**Results.** HMDB51Clustering v_measure for timesformer-base-finetuned-k400: 0.7287.

TimeSformer prepends a CLS token and its classification head reads last_hidden_state[:, 0], so the wrapper pools the same way rather than averaging.

I use AutoImageProcessor rather than AutoVideoProcessor, since TimesformerVideoProcessor only exists in transformers 5.x while mteb supports 4.40 and up. It also matters for correctness: that class defaults to ImageNet mean and std, while every checkpoint config declares VideoMAEImageProcessor with its own values. Frames are passed as HWC lists because a list of 4D tensors trips make_batched on the v4 slow processor.

num_frames comes from config.num_frames, 8 for base and 16 for hr, since the temporal position embeddings are not interpolated.

Every published TimeSformer checkpoint is supervised on Kinetics-400, Kinetics-600, or Something-Something V2; there is no pretrained-only variant, so those tasks are declared in training_datasets. Worth knowing for the leaderboard, since running these on Kinetics400V measures label memorisation.

mteb mock-run passes 4/4 video tasks on all three.

### CODE REVIEW

File: mteb/models/model_implementations/timesformer_models.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            videos = batch["video"]
+            padded = [
+                torch.cat(
+                    [v, v[-1:].expand(self.num_frames - v.shape[0], *v.shape[1:])],
+                    dim=0,
+                )
+                if v.shape[0] < self.num_frames
+                else v[: self.num_frames]
+                for v in videos
+            ]
```
Comment (Samoed) [inline mteb/models/model_implementations/timesformer_models.py]: Why padding is needed? This won't by done by processor?

File: mteb/models/model_implementations/timesformer_models.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    model_type=["dense"],
+    citation=_TIMESFORMER_CITATION,
+    contacts=None,
+    output_dtypes=None,
+    extra_requirements_groups=None,
+)
+
+timesformer_hr_k400 = ModelMeta(
+    loader=TimesformerWrapper,
+    name="facebook/timesformer-hr-finetuned-k400",
```
Comment (Samoed) [inline mteb/models/model_implementations/timesformer_models.py:188]: These models tuned for classification and I don't think this is worth to adding them

File: mteb/models/model_implementations/timesformer_models.py (hunk 1 above)
Comment (hubielu) [reply mteb/models/model_implementations/timesformer_models.py]: You're right that the extra padding isn't needed. FramesCollator(num_frames=self.num_frames) already selects exactly the required number of frames, so I removed the redundant padding/truncation before the processor.

File: mteb/models/model_implementations/timesformer_models.py (hunk 2 above)
Comment (hubielu) [reply mteb/models/model_implementations/timesformer_models.py:188]: Yeah, all three available TimeSformer checkpoints are classification-finetuned. Would you prefer that I drop the TimeSformer addition entirely?

Comment (hubielu) [reply mteb/models/model_implementations/timesformer_models.py:188]: @Samoed Hi Samoed, just wanted to follow up on this one

Comment (Samoed) [reply mteb/models/model_implementations/timesformer_models.py:188]: Yes, I don't think that we want to add it
